### PR TITLE
PC-502: Allow service managers to download referrals

### PR DESCRIPTION
**Pull request by AyaPK**
- Adds two buttons to the suppliers page
  - One to download all referrals
  - One to download referrals made in the last week
- They download in excel format
- They omit all customer PII

### CODE REVIEW

File: tests/test_frontdoor.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def test_british_gas_unavailable():
     page = page.click(contains="Back")
     assert page.has_one("h1:contains('Select your home energy supplier from the list below')")
 
+
```
Comment (AyaPK) [inline tests/test_frontdoor.py:103]: idk the linter did this

File: help_to_heat/portal/download_views.py [hunk 2]
```diff
@@ -78,6 +105,38 @@ def create_referral_csv(referrals, file_name):
     return response
 
 
+def create_referral_all_xlsx(referrals, file_name):
```
Comment (AyaPK) [inline help_to_heat/portal/download_views.py:108]: in hindsight this is probably better named as something like `create_referral_xlsx_no_pii`

File: help_to_heat/portal/download_views.py [hunk 3]
```diff
@@ -138,6 +197,26 @@ def download_feedback_view(request):
...
+    downloaded_at = timezone.now()
+    file_name = downloaded_at.strftime("%d-%m-%Y %H_%M")
+    response = create_referral_all_xlsx(feedbacks, file_name)
+    return response
+
+
+@require_http_methods(["GET"])
+@decorators.requires_service_manager
+def download_referrals_last_week_view(request):
+    feedbacks = portal_models.Referral.objects.filter(created_at__gte=datetime.now() - timedelta(days=7))
```
Comment (AyaPK) [inline help_to_heat/portal/download_views.py:213]: I only discovered this syntax today and it's SO COOL!?

Comment (CheshireSwift) [reply help_to_heat/portal/download_views.py:213]: The filter syntax? It's pretty good, yeah :D

Comment (AyaPK) [reply help_to_heat/portal/download_views.py:213]: Oh specifically the date filtering here!

Comment (CheshireSwift) [reply help_to_heat/portal/download_views.py:213]: Yeah, time deltas are fun.

File: help_to_heat/portal/download_views.py [hunk 4]
```diff
@@ -138,6 +197,26 @@ def download_feedback_view(request):
     return response
 
 
+@require_http_methods(["GET"])
+@decorators.requires_service_manager
+def download_referrals_all_view(request):
+    feedbacks = portal_models.Referral.objects.all()
```
Comment (chidin194) [inline help_to_heat/portal/download_views.py:203]: Was just looking at the PR for general learning, but I think this variable might want to be named `referrals` as opposed to `feedbacks` in this function and the one below?

**Review by davidm-m — CHANGES_REQUESTED** [review]
Mostly looks good, I think we've got some good opportunities for commonisation though!

**Review by davidm-m — APPROVED** [review]
Amazing! Those methods are looking much nicer.
One question before you merge: what makes these methods wet??

**Comment by AyaPK** [comment]
> Amazing! Those methods are looking much nicer. One question before you merge: what makes these methods wet??

I wouldn't say that they are anymore, but previously we had methods that basically did the same thing and a bunch of repeated code. (Admittedly I might be using terminology wrong in this commit!)

**Pull request merged**